include/fraction.h

```diff
@@ -0,0 +1,172 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2023-2023  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_FRACTION_H
+#define DOSBOX_FRACTION_H
+
+#include <cassert>
+#include <numeric>
+
+// Class to represent simple fractions. The fraction is always simplified
+// after construction or after any operation. The sign always is normalised so
+// the numerator carries the sign and the denominator is always positive.
+class Fraction {
+public:
+	constexpr Fraction()
+	{
+		num   = 0;
+		denom = 1;
+	}
+
+	constexpr Fraction(const int64_t n) : num(n), denom(1) {}
+
+	constexpr Fraction(const int64_t n, const int64_t d) : num(n), denom(d)
+	{
+		assert(d != 0);
+
+		if (n == 0) {
+			assert(num == 0);
+			denom = 1;
+			return;
+		}
+		// Simplify fraction
+		const auto gcd = std::gcd(num, denom);
+		num /= gcd;
+		denom /= gcd;
+
+		// Normalise sign so the denominator is always positive
+		if (denom < 0) {
+			num   = -num;
+			denom = -denom;
+		}
+	}
+
+	constexpr int64_t Num() const
+	{
+		return num;
+	}
+
+	constexpr int64_t Denom() const
+	{
+		return denom;
+	}
+
+	constexpr Fraction Inverse() const
+	{
+		return Fraction(denom, num);
+	}
+
+	constexpr double ToDouble() const
+	{
+		return static_cast<double>(num) / denom;
+	}
+
+	constexpr double ToFloat() const
+	{
+		return static_cast<float>(num) / denom;
+	}
```

Comment:
Should return `float`.

Compilers will probably warn of loss of precision because float can't fully represent an int64_t (`denom` probably needs to be cast to float too)

Reply:
Done.